math/src/elliptic_curve/short_weierstrass/curves/bls12_381/twist.rs

```diff
@@ -109,8 +95,7 @@ impl ShortWeierstrassProjectivePoint<BLS12381TwistCurve> {
         let seed_times_p = self.operate_with_self(SEED);
         let psi_plus_seed_times_p = self.psi().operate_with(&seed_times_p);
 
-        Degree2ExtensionField::is_zero(psi_plus_seed_times_p.z().value())
-            && BLS12381TwistCurve::is_on_curve(&psi_plus_seed_times_p)
+        psi_plus_seed_times_p.z() == &FieldElement::zero()
```

Comment:
points have a `is_neutral_element` already implemented with this logic. You can use that here too.